packages/utils/rbac.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react';
+import {
+  AccessReviewResourceAttributes,
+  k8sCreate,
+  K8sVerb,
+} from '@openshift-console/dynamic-plugin-sdk';
+import * as _ from 'lodash-es';
+import log from 'loglevel';
+import { useSafetyFirst } from '../components/safety-first';
+import { ProjectModel, SelfSubjectAccessReviewModel } from '../models';
+import { SelfSubjectAccessReviewKind } from '../types';
+
+// Memoize the result so we only make the request once for each access review.
+// This does mean that the user will have to refresh the page to see updates.
+// Accept an `impersonateKey` parameter to include in the cache key even though
+// it's not used in the function body. (Impersonate headers are added
+// automatically by `k8sCreate`.) This function takes in the destructured
+// resource attributes so that the cache keys are stable. (`JSON.stringify` is
+// not guaranteed to give the same result for equivalent objects.)
+const checkAccessInternal = _.memoize(
+  (
+    group: string,
+    resource: string,
+    subresource: string,
+    verb: K8sVerb,
+    name: string,
+    namespace: string,
+    // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+    impersonateKey: string
+  ): Promise<SelfSubjectAccessReviewKind> => {
+    // Projects are a special case. `namespace` must be set to the project name
+    // even though it's a cluster-scoped resource.
+    const reviewNamespace =
+      group === ProjectModel.apiGroup && resource === ProjectModel.plural
+        ? name
+        : namespace;
+    const ssar: SelfSubjectAccessReviewKind = {
+      apiVersion: 'authorization.k8s.io/v1',
+      kind: 'SelfSubjectAccessReview',
+      spec: {
+        resourceAttributes: {
+          group,
+          resource,
+          subresource,
+          verb,
+          name,
+          namespace: reviewNamespace,
+        },
+      },
+    };
+    return k8sCreate({ model: SelfSubjectAccessReviewModel, data: ssar });
+  },
+  (...args) => [...args].join('~')
+);
+
+const useAccessReview = (
```

Comment:
```suggestion
export const useAccessReview = (
```

if anyone wants to use `loading` value as well...

Reply:
@SanjalKatiyar Here I' m following the [YAGNI principle](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it): for now there is no single use case for this...

Reply:
> @SanjalKatiyar Here I' m following the [YAGNI principle](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it): for now there is no single use case for this... Ideally, we should export it only when there is a need for it.

Reply:
btw we have a use case for `loading` in one of the Jira which is in-progress right now, maybe we can make the change there...